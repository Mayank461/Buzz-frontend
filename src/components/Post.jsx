import React, { useState, useEffect } from "react";

export default function Post({ index, data, inclike, deslike, commentBox, userdata }) {
  let { post_url, _id, like, dislike, comment, post_caption } = data;
  let { firstname, lastname, picture_url } = data.posted_by;
  const [commentmessage, setcommentmessage] = useState();
  const oninputchange = (e) => {
    setcommentmessage({ ...commentmessage, [e.target.name]: e.target.value });
  };
  useEffect(() => {
    console.log(userdata.picture_url);

  });

  return (
    <>
      <div
        key={index}
        className="card p-3 mb-3 shadow p-3 mb-5 bg-body rounded border-0"
      >
        <div className="d-flex justify-content-between align-items-center mb-2">
          <div className="d-flex align-items-center">
            {picture_url ? (
              <img
                src={picture_url}
                className="card-img-top small-round-pic  round-img"
                alt="..."
              />
            ) : (
              <i className="fa-solid fa-user fa-2x card-img-top small-round-pic  round-img bg-warning d-flex justify-content-center align-items-center"></i>
            )}

            {/* <img
                          src={.picture_url}
                          className="card-img-top small-round-pic  round-img"
                          alt="..."
                        /> */}

            <div className="ms-2">{firstname + " " + lastname}</div>
          </div>

          <div className="d-flex align-items-center ms-2 round-img border rounded-circle icon-bg text-dark p-2">
            <i className="fa-solid fa-ellipsis"></i>
          </div>
        </div>
        <div className="ms-2 mb-2">{post_caption}</div>
        <img src={post_url} className="card-img-top rounded-3" alt="..." />

        <div className="d-flex justify-content-between mt-2">
          <div className="d-flex">
            <div>
              <i className="fa-solid fa-thumbs-up bg-primary round-img text-white p-1 me-2"></i>
              {like.length}
            </div>
            <div>
              <i className="fa-solid fa-thumbs-down bg-primary round-img text-white p-1 me-2"></i>
              {dislike.length}
            </div>
          </div>
          <div>1 commment</div>
        </div>

        <div className="d-flex justify-content-between mt-3 border-top border-bottom p-2">
          <button
            onClick={() => {
              inclike(_id);
            }}
          >
            <i className="fa-regular fa-thumbs-up me-2"></i>Like
          </button>
          <button
            onClick={() => {
              deslike(_id);
            }}
          >
            <i className="fa-regular fa-thumbs-down me-2"></i>Dislike
          </button>
          <button
            onClick={() => {
              commentBox(_id, commentmessage);
            }}
          >
            <i className="fa-regular fa-message me-2"></i>Comment
          </button>
        </div>
        <div className="d-flex mt-3">
        {userdata.picture_url ? (
                  <img
                    src={userdata.picture_url}
                    className="card-img-top small-round-pic me-2  round-img"
                    alt="..."
                  />
                ) : (
                  <i className="fa-solid fa-user fa-2x card-img-top small-round-pic  round-img bg-warning d-flex justify-content-center align-items-center"></i>
                )}
          <input
            type="text"
            className="form-control rounded-pill"
            id="comment-box"
            placeholder="Write a comment..."
            onChange={(e) => oninputchange(e)}
            name="message"
          />
        </div>
        {comment.map((element) => {
          return (
            <div className="d-flex align-items-center">
              <div>
                {element.picture_url ? (
                  <img
                    src={element.picture_url}
                    className="card-img-top small-round-pic me-2  round-img"
                    alt="..."
                  />
                ) : (
                  <i className="fa-solid fa-user fa-2x card-img-top small-round-pic  round-img bg-warning d-flex justify-content-center align-items-center"></i>
                )}
              </div>
              <div>{element.message}</div>
            </div>
          );
        })}
      </div>
    </>
  );
}
