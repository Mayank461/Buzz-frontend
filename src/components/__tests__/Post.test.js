import { render, screen } from '@testing-library/react';
import Post from "../Post";
test('should render Post Panel', () => {
  const index = "";
  const element = { posted_by: { firstname: "Mayank", lastname: "Jaggi" }, like: [{}], dislike: [{}], comment: [{}] };
  const like = "";
  const dislike = "";
  const commentBox = "";
  const userData = { _id: "56789" };
  const report = "";
  render(<Post
    index={index}
    data={element}
    inclike={like}
    deslike={dislike}
    commentBox={commentBox}
    userdata={userData}
    reportPost={report}
    uid={userData._id}
  />)
  const text = screen.queryByTestId('whoPosted').innerHTML;
  expect(text).toBe("Mayank Jaggi");

})