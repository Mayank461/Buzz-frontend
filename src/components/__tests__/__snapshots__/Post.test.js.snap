// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Post Panel 1`] = `
<div
  className="card p-3 mb-3 shadow p-3  bg-body rounded border-0"
>
  <div
    className="d-flex justify-content-between align-items-center mb-2"
  >
    <div
      className="d-flex align-items-center"
    >
      <i
        className="fa-solid fa-user fa-2x card-img-top small-round-pic  round-img text-success d-flex justify-content-center align-items-center"
        style={
          Object {
            "backgroundColor": "#F0F2F5",
          }
        }
      />
      <div
        className="ms-2 fw-bold"
        data-testid="whoPosted"
      >
        mahir asrani
      </div>
    </div>
    <div
      className="pointer"
    >
      <i
        className="fa-solid fa-ellipsis "
        data-bs-toggle="dropdown"
        id="dropdownMenuButton1"
      />
      <ul
        aria-labelledby="dropdownMenuButton1"
        className="dropdown-menu"
      >
        <li>
          <div
            className="dropdown-item"
            data-bs-target="#exampleModal"
            onClick={[Function]}
          >
            Report
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="ms-2 mb-2"
  />
  <div
    className="d-flex justify-content-between mt-2"
  >
    <div
      className="d-flex"
    >
      <div
        className="me-2"
      >
        <i
          className="fa-solid fa-thumbs-up bg-primary round-img text-white p-1 me-2"
        />
        1
      </div>
      <div>
        <i
          className="fa-solid fa-thumbs-down bg-danger round-img text-white p-1 me-2"
        />
        1
      </div>
    </div>
    <div
      className="me-1"
    >
      1
       comments
    </div>
  </div>
  <div
    className="d-flex justify-content-between mt-3 border-top border-bottom p-2"
  >
    <div
      className="pointer"
      onClick={[Function]}
    >
      <i
        className="fa-regular fa-thumbs-up me-2"
      />
      Like
    </div>
    <div
      className="pointer"
      onClick={[Function]}
    >
      <i
        className="fa-regular fa-thumbs-down me-2"
      />
      Dislike
    </div>
    <div
      className="pointer"
      onClick={[Function]}
    >
      <i
        className="fa-regular fa-message me-2"
      />
      Comment
    </div>
  </div>
  <div
    id=""
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className=" mt-3 "
    >
      <div
        className="row align-items-center"
      >
        <div
          className="col-12 px-2 mb-2 d-flex align-items-center"
        >
          <i
            className="fa-solid fa-user fa-2x card-img-top small-round-pic  round-img bg-warning d-flex justify-content-center align-items-center"
          />
          <div
            className="input-group"
          >
            <input
              className="form-control"
              name="message"
              onChange={[Function]}
              placeholder="Write a comment..."
              style={
                Object {
                  "borderBottomLeftRadius": "50px",
                  "borderTopLeftRadius": "50px",
                }
              }
              type="text"
            />
            <div
              className="input-group-append"
            >
              <button
                className="btn btn-outline-darkgit  "
                onClick={[Function]}
                type="button"
              >
                <i
                  className="fa-solid fa-2x fa-paper-plane text-success"
                  style={
                    Object {
                      "fontSize": "1.2rem",
                    }
                  }
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mb-3"
    >
      <div
        className="d-flex container"
      >
        <i
          className="mt-2 me-2 fa-solid fa-user fa-2x card-img-top small-round-pic  round-img bg-warning d-flex justify-content-center align-items-center"
        />
        <div
          className="d-flex flex-column w-100 mt-2"
        >
          <div
            className=" align-self-start text-comment"
            data-testid="cmnt"
          />
          <div
            className="mb-3 replyTo"
            onClick={[Function]}
          >
            Reply
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
